exec/java-exec/src/main/java/org/apache/drill/exec/ops/FragmentStats.java

```diff
@@ -79,4 +71,21 @@ public void addOperatorStats(OperatorStats stats) {
     operators.add(stats);
   }
 
+  //DRILL-6197
+  public OperatorStats addOrReplaceOperatorStats(OperatorStats stats) {
+    //Remove existing stat
+    OperatorStats replacedStat = null;
+    int index = 0;
+    for (OperatorStats opStat : operators) {
```

Comment:
I am worried about the small overheads of doing this linear search adding up for each operator, especially for queries with complex query plans.  The stats collection should ideally impose minimal overhead.  Does the operator stats have to be a list or can just use a Set ?

Reply:
Some TPC-DS queries have fairly long list of operators within a fragment and in general it would be preferable to not do this search.  Can you point to where this Json serialization happens ?  my guess is it just needs to preserve the insertion order.  In that case we could use a LinkedHashSet which would provide both the duplicate removal and keep insertion order.

Reply:
I see your point. Also, digging into the code shows I can substitute with a LinkedHashMap, since the list is only referenced here for consumption of its contents:
https://github.com/kkhatua/drill/blob/65efe3ea0c5777490488d3d56cbdb0cb011b9f33/exec/java-exec/src/main/java/org/apache/drill/exec/ops/FragmentStats.java#L45

I can't use a Set, because I need the Stats object hashed on the operator ID & Type, and not the rest of the contents. I'll refactor and try to confirm nothing else breaks.

Reply:
I added a new commit, but I haven't tested it for performance. Can you take a look, @amansinha100 ?

Reply:
LGTM.  Hopefully it did not break existing stuff..so will wait for your confirmation.

Reply:
Everything worked fine. Tried doing a join for TPCH tables - `lineitem` and `orders`, and confirmed no more duplicates for SCREEN, SINGLE_SENDER and HASH_PARTITION_SENDER. For a smaller substitute of `orders` with `supplier` ; confirmed that the BROADCAST_SENDER was also not having duplicates..